presto-tests/src/test/java/com/facebook/presto/tests/TestQueryManager.java

```diff
@@ -152,7 +152,7 @@ public void testFailQueryPrerun()
         QueryState state = dispatchManager.getQueryInfo(queryId).getState();
         assertEquals(state, FAILED);
         //Give the stats a time to update
-        Thread.sleep(1000);
+        Thread.sleep(5000);
         assertEquals(queryManager.getStats().getQueuedQueries(), 0);
```

Comment:
The better way is to move `getQueuedQueries() == 0` check into the `while` block. The `while` block should also have a timeout, instead of `true`.

Reply:
Done